corehq/apps/data_interfaces/deduplication.py

```diff
@@ -4,16 +4,63 @@
 
 from corehq.apps.case_search.const import SPECIAL_CASE_PROPERTIES_MAP
 from corehq.apps.data_interfaces.utils import iter_cases_and_run_rules
-from corehq.apps.es import CaseES, queries
+from corehq.apps.es import queries
 from corehq.apps.es.case_search import CaseSearchES
 from corehq.messaging.util import MessagingRuleProgressHelper
 
 DUPLICATE_LIMIT = 1000
 DEDUPE_XMLNS = 'http://commcarehq.org/hq_case_deduplication_rule'
 
 
-def find_duplicate_case_ids(domain, case, case_properties, include_closed=False, match_type="ALL"):
-    es = CaseSearchES().domain(domain).size(DUPLICATE_LIMIT).case_type(case.type)
+def _get_es_query(domain, case, case_filter_criteria=[]):
+    # Import here to avoid circular import error
+    from corehq.apps.data_interfaces.models import (
+        MatchPropertyDefinition,
+        LocationFilterDefinition
+    )
+
+    query = CaseSearchES().domain(domain).size(DUPLICATE_LIMIT).case_type(case.type)
+
+    def apply_criterion_to_query(query_, definition):
+        if isinstance(definition, MatchPropertyDefinition):
+            match_type = definition.match_type
+
+            if match_type == MatchPropertyDefinition.MATCH_HAS_NO_VALUE:
+                query_ = query_.case_property_has_no_value(definition.property_name)
+            elif match_type == MatchPropertyDefinition.MATCH_HAS_VALUE:
+                query_ = query_.case_property_has_value(definition.property_name)
```

Comment:
I'd just do this, instead of defining a new method
```suggestion
                query_ = query_.NOT(case_property_missing(definition.property_name))
```
or you could make a `case_property_exists` filter function and use that.